MetaMask/metamask-mobile
# fix: 942 invalid QR code warning message on bad QR code

## **Description**
When trying to scan an invalid QR code (e.g. not an address) from the home screen, the QR scanner was scanning and then just closing without providing any feedback to the user. This PR adds a warning to the user that they have scanned an invalid QR code.

## **Related issues**

Fixes: [942](https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/gh/metamask/mobile-planning/942)

## **Manual testing steps**

**Steps to test:**
1. On the Main Wallet view tap the Scan Icon on the top right
2. Scan this invalid QR code https://www.screencast.com/t/qhiE60jx on Networks , Ethereum, Polygon, Binance and Avalanche.
3. Should see a popup warning that the QR code is invalid

**Regression test:**
1. Scan a valid QR address code
2. Should go to SendTo flow with address populated

## **Screenshots/Recordings**



### **Before**


https://github.com/MetaMask/metamask-mobile/assets/141057783/1369c8ea-09a3-43ed-b2e6-3dcd3c4a5eb6



### **After**


https://github.com/MetaMask/metamask-mobile/assets/141057783/a432bd4f-ddad-4ca1-ac60-5093f6ed4731



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
